### PR TITLE
refactor: extract command-validation layer for CLI boundary checks

### LINKED ISSUE #218: refactor: validate CLI string inputs into strong types at the application boundary
## Problem

Raw string inputs from CLI arguments and flags are currently passed deep into command implementations, where they are narrowed to SDK types via `as T` casts (e.g. `as OwnerTypeEnum`, `as AppName`, `as OutputMode`). This spreads validation logic across the codebase and means the internals of the application operate on unvalidated strings rather than fully type-safe values.

## Proposal

Validate all CLI string inputs into their strong types **at the application boundary** (the parseArgs/routing layer in `src/index.ts`), so that command implementations receive already-validated, correctly-typed values. This would:

1. **Eliminate `as T` casts** in command files — the type system enforces correctness internally
2. **Centralise validation** — invalid inputs are rejected early with clear error messages
3. **Make the application interior fully type-safe** — no runtime type mismatches possible past the boundary

### Concrete examples

| CLI input | Current (cast in impl) | Target (validated at boundary) |
|---|---|---|
| `--ownerType` | `options.ownerType as OwnerTypeEnum` | Validated via runtime enum values from SDK |
| `--resourceType` | `options.resourceType as ResourceTypeEnum` | Validated via runtime enum values from SDK |
| `--permissions` | `.filter(...) as PermissionTypeEnum[]` | Each value validated against SDK enum |
| `open <app>` | `app as AppName` | Validated via `isAppName()` type guard |
| `--output` | `as OutputMode` | Validated at boundary |

### Prerequisite

The JS SDK (`@camunda8/orchestration-cluster-api`) needs to export concrete enum values at runtime (not just type aliases) so that type guards can validate CLI strings against the actual set of valid values without duplicating them.

## Relationship to #213

Issue #213 addresses command argument and flag **structural** validation (required flags, argument counts, etc.). This issue is about **semantic** validation — ensuring string values are valid members of their target types before entering the application's typed interior.

Together they form a complete input validation strategy at the CLI boundary.

### PR DESCRIPTION
**Pull request by jwulf**
## Summary

Introduces a **command-validation architectural layer** (`src/command-validation.ts`) that validates and type-narrows CLI string inputs before they reach command handlers. Handlers receive already-validated, properly-typed inputs — no internal validation needed.

Refactors both `identity-authorizations` and `open` commands to use boundary validation.

Closes #218

## Architecture

```
parseArgs()
  ↓
#213 (future): requireResource(verb, resource)      ← dispatch guard
  ↓
#212 (future): validateAcceptedFlags(values, ...)    ← dispatch guard
  ↓
#218 (this PR): validateXxxOptions(raw)              ← per-command validation
  ↓
handler(validated)                                   ← implementation
```

Each layer uses utilities from `command-validation.ts`. The module is designed as the extension point for #212 and #213.

## `src/command-validation.ts` — Boundary validation utilities

| Function | Purpose | Extension point |
|---|---|---|
| `requireOption(value, flag)` | Validate required `--flag` presence | Used by all commands |
| `requireEnum<T>(value, enumObj, flag)` | Validate + narrow to SDK enum type | #218 |
| `requireCsvEnum<T>(value, enumObj, flag)` | Validate + narrow CSV to enum array | #218 |
| `requirePositional(value, label, hint?)` | Validate required positional arg | #218 |
| `requireOneOf<T>(value, allowed, label, hint?)` | Validate + narrow to tuple member | #218 |
| *(future)* `validateAcceptedFlags()` | Detect unknown flags per verb×resource | #212 |
| *(future)* `requireResource()` | Validate resource arg before dispatch | #213 |

### Zero `as T` casts

Uses `Object.values(enumObj).find(v => v === value)` for enum validation and `allowed.find(v => v === value)` for tuple validation. Both return `T | undefined`. After `process.exit(1)` on `undefined`, TypeScript narrows to `T`. No casts, no `biome-ignore` suppressions anywhere.

### `src/command-validation.ts` (new)
- `requireOption()` — exits with `--flag is required`
- `requireEnum()` — exits with `Invalid --flag "X". Valid values: ...`
- `requireCsvEnum()` — exits with `Invalid --flag: X, Y. Valid values: ...`
- `requirePositional()` — exits with `Label is required` + optional usage hint
- `requireOneOf()` — exits with `Unknown label 'X'. Available: ...` + optional usage hint

### `src/commands/identity-authorizations.ts`
- **Removed**: 3 inline type guard functions (`isOwnerType`, `isResourceType`, `isPermissionType`)
- **Removed**: all validation logic from `createIdentityAuthorization` handler
- **Removed**: 3 `biome-ignore lint/plugin:` suppressions for `as T` casts
- **Added**: `CreateAuthorizationInput` interface with validated types
- **Added**: `validateCreateAuthorizationOptions()` — raw options → validated input
- **Changed**: `createIdentityAuthorization()` now accepts `CreateAuthorizationInput` (non-optional, typed fields)

### `src/commands/open.ts`
- **Removed**: inline `if (!app)` / `if (!isAppName(app))` validation from handler
- **Added**: `OpenAppInput` interface with validated types
- **Added**: `validateOpenAppOptions()` — raw positional + options → validated input
- **Changed**: `openApp()` now accepts `OpenAppInput` (single validated object)

### `src/index.ts`
- Dispatch points now use two-step validate-then-handle pattern:
  - `validateCreateAuthorizationOptions()` → `createIdentityAuthorization(validated)`
  - `validateOpenAppOptions()` → `openApp(validated)`

#### `tests/unit/command-validation.test.ts` (new) — 23 tests
- Direct unit tests for all 5 validation utilities
- Covers: valid input passthrough, missing/empty rejection, invalid enum/value rejection, case sensitivity, CSV edge cases (whitespace, trailing commas), hint output

#### `tests/unit/identity.test.ts` — updated
- Validation tests now call `validateCreateAuthorizationOptions()` directly (sync `assert.throws`)
- Added enum validation tests: invalid `--ownerType`, `--resourceType`, `--permissions` rejected with valid-values listing
- Added `--resourceType` required-field test (was missing)
- Added positive test: valid enum values pass through correctly

#### `tests/unit/identity-authorization-cli.test.ts` (new) — 8 tests
- CLI behavioral tests spawning subprocess with `--dry-run`
- Covers: body shape, CSV trimming, enum rejection (ownerType, resourceType, permissions), missing-field rejection, DELETE URL shape

#### `tests/unit/open.test.ts` — updated
- Updated `openApp` tests to call `validateOpenAppOptions()` (where validation now lives)
- Added positive test: all valid app names accepted
- Existing CLI integration tests with `--dry-run` all pass (7 tests: all 4 apps, invalid app, missing app, Cloud URL)

## Verification

- `npx tsc --noEmit` — clean
- `npx biome check src/ tests/` — clean (zero suppressions, zero diagnostics)
- `node --test` — 825 tests, 824 pass, 0 fail, 1 skipped

### CODE REVIEW

File: src/command-validation.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	value: string,
+	enumValues: Record<string, T>,
+	flagName: string,
+): T[] {
+	const validValues = Object.values(enumValues);
+	const items = value
+		.split(",")
+		.map((p) => p.trim())
+		.filter((p) => p.length > 0);
+
```
[the remark this answers is not in the document]
Comment (jwulf) [reply src/command-validation.ts:70]: Fixed in 072c144 — added `items.length === 0` guard that exits with `--{flag} is required` when input is only commas/whitespace. Test added for this edge case.

File: tests/unit/open.test.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import assert from 'node:assert';
 import { spawnSync } from 'node:child_process';
 import { join } from 'node:path';
 import { tmpdir } from 'node:os';
-import { deriveAppUrl, getBrowserCommand, openApp, OPEN_APPS } from '../../src/commands/open.ts';
+import { deriveAppUrl, getBrowserCommand, openApp, validateOpenAppOptions, OPEN_APPS } from '../../src/commands/open.ts';
```
[the remark this answers is not in the document]
Comment (jwulf) [reply tests/unit/open.test.ts]: Fixed in 072c144 — removed the unused `openApp` import.
